README.md

```diff
@@ -13,175 +13,355 @@ Compartmentalization of host behavior hooks for JS.
-type ModuleNamespace = object;
-// Used to create a Reusable Instance factory for Module Records
-// exotic
-interface StaticModuleRecord {
-  // intend to add reflection of import/export bindings
+interface ModuleExportsNamespace {}
+
+// Module instances have a module environment record, an exotic object
+// that reflects every name in the lexical scope of the module.
+// The environment record does not contain a property for any names that are
+// imported and reexported without a lexical binding.
+// An `import name as alias` binding will have a property with the lexically bound alias. 
+// An `export name as alias` binding will have a property with the lexically bound name,
+// whereas the module exports namespace will have a property with the alias.
+interface ModuleEnvironmentRecord {}
+
+// Bindings reflect the `import` and `export` statements of a module.
+type Binding =
+  { import: '*' | string, as?: string, from: string } |
+  { export: '*' | string, as?: string, from?: string };
+
+// Compartments support ECMAScript modules and linkage to other kinds of modules,
+// notably allowing for JSON or WASM.
+// These must provide an initializer function and may have bindings.
+// The bindings correspond to the equivalent `import` and `export` declarations
+// of an ECMAScript module.
+type ThirdPartyStaticModuleRecord = {
+  bindings?: Array<Binding>,
+  // Initializes the module if it is imported.
+  // Initialize may return a promise, indicating that the module uses
+  // the equivalent of top-level-await.
+  initialize(environment: ModuleEnvironmentRecord, meta: Object),
+  // TODO decide upon a positive or negative form of `usesMeta` or `noMeta`.
+  // Avoid calling `importMetaHook` for this module since it won't use the
+  // `meta` property. 
+  noMeta?: boolean,
+  // TODO consider whether to reflect top-level-await statically.
+};
 
-  // needs to allow duplicates
-  staticImports(): {specifier: string, exportNames}[];
-}
-interface SourceTextStaticModuleRecord extends StaticModuleRecord {
-  // no coerce
-  constructor(source: string);
+// Static module records are an opaque token representing the compilation
+// of a module that can be reused across multiple compartments.
+interface StaticModuleRecord {
+  // Static module records can be constructed from source.
+  // XS allows third-party module records and source descriptors to
+  // be precompiled as well.
+  constructor(source: string | { source: string } | ThirdPartyStaticModuleRecord);
+
+  // Static module records reflect their bindings for information only.
+  // Compartments use internal slots for the compiled code and bindings.
+  bindings: Array<Binding>;
 }
+
+// A ModuleDescriptor captures a static module record and per-compartment metadata.
+type ModuleDescriptor = 
+  // Using a string for a module descriptor indicates a static module record
+  // should be inherited from the parent compartment, but not the parent
+  // compartment's instance.
+  string
```

Comment:
@mhofman suggests that we should make module descriptors homogeneously objects.
```suggestion
  // Indicates a static module record
  // should be inherited from the parent compartment, but not the parent
  // compartment's instance.
  {
    inherit: string,
  }
```

Reply:
We (@erights, @patrick-soquet, @phoddie, @mhofman, myself) did not come to a conclusion either for “whether” or “with what name”.